web/react/components/edit_channel_purpose_modal.jsx

```diff
@@ -90,7 +90,7 @@ export default class EditChannelPurposeModal extends React.Component {
                     </Modal.Title>
                 </Modal.Header>
                 <Modal.Body>
-                    <p>{`Describe how this ${Utils.getChannelTerm(this.props.channel.channelType)} should be used.`}</p>
+                    <p>{`Describe how this ${Utils.getChannelTerm(this.props.channel.channelType)} should be used. This text appears in the channel list in the "More..." menu and helps other decide whether to join.`}</p>
```

Comment:
Without the location it is unclear where in the UI the purpose is used. I think if we change anything in the future we can update the help text again, but we should still try to keep it accurate and helpful for now.